File: 349.intersection-of-two-arrays.java
/*
 * @lc app=leetcode id=349 lang=java
 *
 * [349] Intersection of Two Arrays
 */

// @lc code=start

import java.util.HashSet;
import java.util.Set;

class Solution {
    public int[] intersection(int[] nums1, int[] nums2) {
        Set<Integer> set1 = new HashSet<>(), set2 = new HashSet<>();
        for (int n : nums1) {
            set1.add(n);
        }
        for (int n : nums2) {
            set2.add(n);
        }
        set1.retainAll(set2);
        return set1.stream().mapToInt(x -> x).toArray();
    }
}
// @lc code=end

File: 349.intersection-of-two-arrays.cpp
/*
 * @lc app=leetcode id=349 lang=cpp
 *
 * [349] Intersection of Two Arrays
 */

// @lc code=start

#include <algorithm>
#include <set>
#include <vector>

using namespace std;

class Solution {
public:
    vector<int> intersection(vector<int>& nums1, vector<int>& nums2) {
        set<int> set1{nums1.begin(), nums1.end()}, set2{nums2.begin(), nums2.end()};
        vector<int> ret;
        set_intersection(set1.begin(), set1.end(), set2.begin(), set2.end(), back_inserter(ret));
        return ret;
    }
};
// @lc code=end
